### ISSUE TITLE
mirrord not working with kubie

### ISSUE THREAD

**Issue by tizz98** [metalbear-co/mirrord#616]
### Bug Description

I use [kubie](https://github.com/sbstp/kubie) to manage all of my kubernetes contexts. mirrord seems to respect this value when creating the mirrod-agent pod, but doesn't respect it when trying to fetch the portforwards for the pod.

In the attached logs you can see that mirrord is trying to connect to my local minikube rather than the remote EKS cluster.

### Steps to Reproduce

1. Setup minikube
2. Setup another k8s context
3. Setup kubie
4. Setup mirrord
5. Use `kubie ctx` to change contexts to the other k8s context
6. Attempt to use `mirrord` to exec, but get a 404

### Relevant Logs

```shell
2022-10-21T19:52:10.437214Z DEBUG ThreadId(17) HTTP: kube_client::client::builder: exit http.method=GET http.url=https://192.168.39.39:8443/api/v1/namespaces/default/pods/mirrord-agent-oj5jca1wi3-wmdvh/portforward?&ports=31596 otel.name="HTTP" otel.kind="client" http.status_code=404 otel.status_code="ERROR"
```

### Your operating system and version

Linux jedi 5.19.0-76051900-generic #202207312230~1663791054~22.04~28340d4 SMP PREEMPT_DYNAMIC Wed S x86_64 x86_64 x86_64 GNU/Linux

### Local process

python/hypercorn

### Local process version

python 3.9.11 / hypercorn 0.13.2

### Additional Info

Discord thread: https://discord.com/channels/933706914808889356/950020257156833340/threads/1033082385513861121

We ran into this issue before too, as long as we check `KUBECONFIG` env var then it works as expected:

after running kubie ctx:
```
KUBECONFIG=/tmp/kubie-configEQkU2Q.yaml
KUBIE_KUBECONFIG=/tmp/kubie-configEQkU2Q.yaml
```

**Comment by aviramha** [comment]
Thanks for reporting this.
Small correction - the issue starts when an existing agent is reused (i.e process launched by mirrord forks/execs/creates another process)
We tested if it's related to env variables, seems not (using '-s').

**Comment by meowjesty** [comment]
Hey @tizz98 can you please provide a bit more info about your `kubie` (config, version).

If you have a minimal example app as well it would be greatly appreciated. I've tried to replicate this following your steps but my sample python app worked, so I need a bit more help to debug this further.

**Comment by tizz98** [comment]
```
kubie 0.16.0
```

I'll try to get a minimal example sometime next week

**Comment by aviramha** [comment]
Heyy @tizz98 , we just released 3.4.0 which changed a bit how we use kubeconfig/k8s API and I believe it might solve this issue as well.
Can you please try newest version at your comfort and let us know?

**Comment by tizz98** [comment]
yep that seems to have fixed it! Thanks!
